src/rgw/rgw_rest_swift.cc

```diff
@@ -41,6 +41,110 @@
 
 using namespace std;
 
+template <class H, bool O>
```

Comment:
a comment clarifying what the H and O template parameters mean would be helpful--I mean obviously H is for HMAC.  It took me a long time to find the H actually used at the bottom of the stack of templates.  But it seems reasonable.  I speculate below that it would be clearer if we were able to somehow remove the untemplated versions, that support multiple digest sizes in all their methods...